### PR TITLE
Tf catvars from tlapusan

### PR DESCRIPTION
**Pull request by parrt**
Starting from @tlapusan, raising exceptions when categorical variable's are used in feature space. I just don't know how to display that at the moment. Popping the version to 2.2.0 to incorporate categorical vars. 

Derived from https://github.com/parrt/dtreeviz/pull/263

We still need to test the decision notes more, such as when there are lots of levels in the catvars.

### CODE REVIEW

File: dtreeviz/trees.py [hunk 1]
```diff
@@ -1591,6 +1623,18 @@ def _ctreeviz_bivar(shadow_tree, fontsize, ticks_fontsize, fontname, show,
     _format_axes(ax, shadow_tree.feature_names[featidx[0]], shadow_tree.feature_names[featidx[1]],
                  colors, fontsize, fontname, ticks_fontsize=ticks_fontsize, grid=False)
 
+    if 'splits' in show:
+        plt.draw()
+        print("labels", ax.get_xticklabels(), ax.get_yticklabels())
```
Comment (mepland) [inline dtreeviz/trees.py:1628]: @tlapusan is this print for debugging? Probably should be commented out.

**Review by mepland — CHANGES_REQUESTED** [review]
The code looks good to me, besides one print statement. I don't see text axis tick labels in either of the demo notebooks though? `sex_label` is still 0 or 1.

File: dtreeviz/trees.py (hunk 1 above)
Comment (parrt) [reply dtreeviz/trees.py:1628]: sorry. yeah, debugging.

**Comment by parrt** [comment]
> The code looks good to me, besides one print statement. I don't see text axis tick labels in either of the demo notebooks though? `sex_label` is still 0 or 1.

We have an updated the notebooks yet but I do have tutorial I'm working on that uses categoricals. Take a look at `testing/tf-catvars.py` for some playing around.

I need to fix `testing/tf-catvars2.py`... Working on that now hi cardinality categorical fails.

**Review by mepland — APPROVED** [review]

**Pull request merged**